File: leetcode/go/plus_one.go
func plusOne(digits []int) []int {
    var sum int
    for i := len(digits) -1; i >= 0; i-- {
        sum = digits[i] + 1
        if sum < 10 {
            digits[i] = sum
            return digits
        }
        digits[i] = sum - 10
    }
    return append([]int{1}, digits...)
}

File: leetcode/python/plus_one.py
# python3

class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        result = []
        carry = 1
        for dig in digits[::-1]:
            result.append( (dig + carry) % 10 )
            carry = (dig + carry) // 10
        if carry:
            result.append(1)
        return result[::-1]
